### ISSUE TITLE
LVGL: replace stack-allocated event_head linked list with bounded array stack

### ISSUE THREAD

**Issue by prestonbrown** [prestonbrown/helixscreen#907]
## Background

LVGL's event-dispatch linked list (`event_head` in `src/misc/lv_event.c`) is built from **stack-allocated** `lv_event_t` nodes:

```c
lv_result_t lv_obj_send_event(lv_obj_t * obj, lv_event_code_t event_code, void * param)
{
    lv_event_t e;          // STACK
    lv_event_push(&e);     // event_head = &e
    event_send_core(&e);
    lv_event_pop(&e);      // event_head = e->prev
}
```

This is structurally fragile: any path that pushes without a matching pop strands `event_head` on a defunct stack frame. The next `lv_event_mark_deleted` call walks the dead pointer and crashes with SIGSEGV/SIGBUS in the inner-loop `e->original_target == target` deref.

**Crash family**: #190, #80, #776, #793, #840, #871, #873, #874, #875, #877, #878, #880, plus bundle `RPHAV9T7` (v0.99.53/AD5X) — the bundle that finally pinpointed the mechanism.

## Short-term mitigation (already shipped post-`RPHAV9T7`)

`patches/lvgl_event_pop_unwind_safe.patch` wraps the local `lv_event_t e` with `__attribute__((__cleanup__(lv_event_pop_if_head)))` in both `lv_obj_send_event` and `lv_event_push_and_send`, plus `-fexceptions` on the lvgl C compile so cleanup actions fire on C++ exception unwind. This closes the C++-throw-escapes-C-frame leak path that was the dominant trigger (combined with [L083]'s `std::thread` EAGAIN throw-and-abort pattern on resource-constrained ARM/MIPS targets).

The cleanup path is idempotent (compares `event_head == &e` before popping), so the normal-path explicit `lv_event_pop` still runs and the cleanup is a no-op except on unwind.

This is a **mitigation, not a fix.** Other ways to strand `event_head` on a dead frame still exist:
- Recursive event dispatch where a callback corrupts `e->prev` via stack-buffer-overflow
- Any future code path that calls `lv_event_push` without a `lv_event_pop`
- Out-of-tree callers that accidentally use the public `lv_event_push`/`pop` directly with mismatched scoping

## Proposed long-term fix

Replace the stack-allocated linked list with a **bounded fixed-size array** of `lv_event_t` slots in `lv_global`:

```c
#define LV_EVENT_STACK_DEPTH 64
typedef struct {
    lv_event_t  slots[LV_EVENT_STACK_DEPTH];
    uint8_t     depth;       // number of active entries
} lv_event_stack_t;
```

`lv_event_push` becomes index increment + memcpy; `lv_event_pop` becomes index decrement; `lv_event_mark_deleted` walks `slots[0 .. depth-1]`. Overflow (depth >= cap) logs `helix_lvgl_anomaly` and refuses the push (returning early from `lv_obj_send_event`). Eliminates the design fragility entirely.

### Tradeoffs

- **Pro**: no stale-stack-pointer class of bug possible
- **Pro**: bounded memory cost (64 × ~32 B ≈ 2 KB per process — singleton stack, no per-thread issue under LVGL's single-thread model)
- **Pro**: cache-friendly walk (contiguous array vs pointer-chased list)
- **Pro**: `lv_event_mark_deleted` no longer needs alignment/depth defensive bail
- **Con**: nontrivial LVGL patch (~50 lines)
- **Con**: hard cap on dispatch depth (64 is plenty — current observed max is ~12 even under stress)
- **Con**: `lv_event_t` must be POD-copyable (already true)
- **Con**: invasive enough that we should land on a release boundary, not a hotfix

### Implementation sketch

1. Replace `event_head` and `e->prev` linkage with `lv_global.event_stack` (array) + `event_stack_depth` (uint8).
2. `lv_event_push(&e)`: if `depth >= cap` → `helix_lvgl_anomaly("event_stack_overflow")` + early-return; else `slots[depth++] = e`.
3. `lv_event_pop(&e)`: `depth--` (assert `&slots[depth] == e_caller_local` only in debug — drop in release).
4. `lv_event_mark_deleted(target)`: `for (i = 0; i < depth; ++i) if (slots[i].original_target == target || slots[i].current_target == target) slots[i].deleted = 1;`
5. Drop the `lv_event.c:340-367` defensive walk; remove `lv_event_pop_if_head`; revert the cleanup-attribute and `-fexceptions` patches (no longer needed).
6. Update LVGL tests; add a test that confirms a thrown C++ exception escaping a callback doesn't corrupt subsequent dispatch.

### Open questions

- Push the patch upstream to lvgl/lvgl, or keep as a vendored patch? Upstream value is high since this affects every LVGL embedder that uses C++. Probably worth a PR conversation.
- Pick `LV_EVENT_STACK_DEPTH` via Kconfig (`lv_conf.h`) or fixed? Fixed at 64 is fine for our targets.
- Confirm no one else uses `lv_event_push`/`pop` outside `lv_obj_send_event` and `lv_event_push_and_send` — quick grep across LVGL says no.

## Priority

Post-1.0 cleanup. Current short-term mitigation is sufficient for the field. Schedule with the LVGL upgrade work or as a focused 1-day refactor.

## References

- Bundle that pinned the mechanism: `RPHAV9T7` (v0.99.53/AD5X)
- Lessons: [L081] `lv_event_mark_deleted` crash signature; [L083] no bare `std::thread().detach()` on ARM/MIPS
- Mitigation patch: `patches/lvgl_event_pop_unwind_safe.patch`
- Defensive walk patch (still useful as belt-and-suspenders): `patches/lvgl_event_mark_deleted_defensive.patch`
- L081 cluster tracker: `~/.claude/projects/-home-pbrown-Code-Printing-helixscreen/memory/project_l081_recurrence_post_840.md`
